module-4/backend/week_one.md

```diff
@@ -1,22 +1,87 @@
 ## Week One - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the 
+majority of what we've learned this week. 
 
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+The most useful thing I leaned over intermission week was how to utlizie the newer ES6 arrow function syntax.
+
 2. What are some tools to help debug JavaScript code?
+
+pryjs, debugger, chrome dev tools
+
 3. What are some tools you need in order to unit test your JavaScript?
+
+Mocha, Chai and Selenium if you are integration testing
+
 4. What is the syntax for invoking a function?
+
+myFunction()
+
 5. What is CORS?
+
+Cross Origin Resource Sharing it is a standard that allows HTTP requests between different origin sites.
+
 6. How do we enable CORS?
+
+We must configure the server to respond to requests from other domains with an Access Control Allow Origin in is preflight response.
+
 7. What's `this` in JavaScript?
+
+This is a self referential objection that refers back to the context in which it is called. 'This' changes value fequently depending on this context.
+Globally, for example, 'this' refers to the window. Within a constructor function 'this' refers to the new object that will be instantiated by the
+contructor. In an Arrow function 'this' will refer to the conext in which the function was invoked. Those are just a few examples of the possible
+meanings of 'this'.
+
 8. What is Webpack and why is it useful?
+
+Webpack is a tool that helps to manage our dependencies with client-side JavaScript. It is useful because it allows use to require single bundled
+JS that create a pipeline for our aset hierarchy as well as delivering minified JS and CSS files to the client browser.
+
 9. When/why do you want to use event delegation?
+
+Event delegation is used when there is a need to add an event listerner to an element that may not initially be loaded on the page. Event delegation
+allows us to bind these event listerners dynamically while protecting us from memory leaks caused by adding event listenrs to elements that are
+not or are no longer on the page.
+
 10. What is a callback function?
+
+A callback function is a higher order function in that it is passed to another function. It is referred to as a callback becuase the initiating
```

Comment:
That's why many Ruby enumerables and JS Array.prototype functions are all higher order, they take a Procs/blocks/functions as arguments.